### ISSUE TITLE
Unexpected end of JSON input

### ISSUE THREAD

**Issue by kapilsriram** [airbnb/javascript#2700]
I am facing similar issue as mentioned in  https://github.com/airbnb/javascript/issues/2540

whenever I run "eslint ." i am thrown "Unexpected end of JSON input"
![image](https://user-images.githubusercontent.com/2054924/209660163-85ac4fd4-05c1-484d-bf14-53972aaf3698.png)

**Comment by ljharb** [comment]
What version of everything are you using? eslint 8.2 is out of date, try with eslint latest, and make sure you're on the latest of `eslint-config-airbnb` as well.

**Comment by kapilsriram** [comment]
i use 19.0.4 of eslint-config-airbnb and v8.30.0 of eslint. Still the same issue.
this is what i have in my eslint.json

extends": ["eslint:recommended","airbnb/whitespace"],

if i replace "airbnb/whitespace" with "airbnb" or even "airbnb/hooks" the issue is resolved. but wont work with whitespace for some reason

**Comment by kapilsriram** [comment]
![image](https://user-images.githubusercontent.com/2054924/209795805-5773b778-069e-4008-821b-fef18bf40240.png)

**Comment by kapilsriram** [comment]
i have 2 different projects, with same issue, in one project, it opens "airbnb/whitespace-async.js" file in my notepad whenever i run "npm run lint" or "eslint ."

**Comment by ljharb** [comment]
The way the `eslint-config-airbnb/whitespace` file works is it executes https://unpkg.com/browse/eslint-config-airbnb@19.0.4/whitespace-async.js - which is an executable node program - and it outputs JSON.

If that file is being opened in your editor then perhaps the executable bit isn't set properly on that file, or perhaps something about your system isn't supporting `execSync` of an executable JS file. It's possible this happens on any Windows machine, but I'd be pretty surprised if you were the first Windows user to use the whitespace config.

If it's a problem on all Windows machines, then we might need to change the approach we use to execute the whitespace-async file.

**Comment by kapilsriram** [comment]
the problem didnt exist in older versions, only in the latest versions we r facing the issue. Previously we used 18.2.0 with typescript 3.9.10

**Comment by ljharb** [comment]
yes, this technique was introduced to be compatible with eslint 8.

**Comment by kapilsriram** [comment]
is it possible to fix this thing? or can give me a hint of how to get around this, I can give a try and probably will contribute the update

**Comment by ljharb** [comment]
I believe that what's required is either finding a package that can sync exec a node file on linux/mac and windows - or, detecting windows, and using an alternative sync approach on windows (all in the whitespace file, to exec the whitespace-async file).

**Comment by kapilsriram** [comment]
![image](https://user-images.githubusercontent.com/2054924/210207773-359f5d91-bb54-4c7c-a75f-87da5c92e62c.png)
by adding "node " to the command the error seized, is this the expected functionality? we are only trying to run the js file in node isnt?

**Comment by ljharb** [comment]
i don't know what you mean by "seized" here - but certainly if running it directly with node fixes the problem, that might work.

**Comment by kapilsriram** [comment]
i mean stopped. cool. I was just checking if "node <command>" does the same thing as it was doing previoulsy

**Comment by ljharb** [comment]
Yes, it should be unnecessary, but it also doesn't hurt. Want to make a PR?

**Comment by kapilsriram** [comment]
yep sure.

**Comment by meta-panic** [comment]
I faced the very same issue recently. So, the problem was the line `#!/usr/bin/env node` [here](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/whitespace-async.js)

My fix:
1. Create/update the symlink at /usr/bin `ln -s <path_to_your_nodejs> /usr/bin/node`
2. Check permission(I just set 777 and that's all)

I hope it'will work for someone else

**Comment by meta-panic** [comment]
@kapilsriram hi! Maybe my workaround could help you

**Comment by Oms-343** [comment]
@meta-panic  could you please elaborate..? I am facing the same issue .

**Comment by Oms-343** [comment]
@kapilsriram your solution helped. Thankyou👍

**Comment by postmeback** [comment]
Looks like the fix worked for many people. 

IMHO, The PR and the issue both can get resolved.

**Comment by zaycker** [comment]
> Yes, it should be unnecessary

sorry for my appearing after years...
it is called "shebang" (Ricky Martin wrote a song about it), and it was also mentioned by @meta-panic 
shebangs don't work under Windows at all, those headers with interpreter definitions starting with `#!`
that's why `node` should be defined explicitly in `execSync` to be cross-platform otherwise it cannot be executed under Windows

> but I'd be pretty surprised if you were the first Windows user to use the whitespace config.

I think people just removed the "airbnb/whitespace" line after they met the issue under Windows (the case of my project - all windows devs have commented the line in the eslint config)

**Comment by thelittleerik** [comment]
fck microsoft how is this still an issue, @kapilsriram your solution fixed it. Thanks

> ![image](https://user-images.githubusercontent.com/2054924/210207773-359f5d91-bb54-4c7c-a75f-87da5c92e62c.png) by adding "node " to the command, the issue seized, is this the expected functionality? we are only trying to run the js file in node isnt?
